tests/Unit/Domain/Test_DomainCreators.cpp

```diff
@@ -211,6 +219,54 @@ void test_physical_separation(
     }
   }
 }
+
+void test_disk_construction(
```

Comment:
move this and other disk specific code to tests/Unit/Domain/DomainCreators/Test_Disk.cpp

Reply:
move the rest into tests/Unit/Domain/DomainTestHelpers.cpp, and only declare the physical_separation in the hpp file

Reply:
Also in the factory creation test, keep the std::unique_ptr<DomainCreator> that is returned by test_factory_creation, dynamic_cast it into a DomainCreators::Disk, and call test_disk_construction.